samplers/probability/consistent/sampler_test.go

```diff
@@ -0,0 +1,251 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package consistent
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type (
+	testDegrees int
+	pValue      int
+
+	testSpanRecorder struct {
+		lock  sync.Mutex
+		spans []sdktrace.ReadOnlySpan
+	}
```

Comment:
Why not use https://pkg.go.dev/go.opentelemetry.io/otel/sdk@v1.5.0/trace/tracetest#NewSpanRecorder?

Reply:
Now using tracetest.InMemoryExporter